### PR TITLE
Exercise 2

### PR DESCRIPTION
**Pull request by steven-mi**

### CODE REVIEW

**Review by MatzeK105 — CHANGES_REQUESTED** [review]
Oh damn, sorry guys!! I reviewed yesterday, but didn't see, that I need to submit the review...

**Comment by steven-mi** [comment]
@medizinmensch We did the Review for https://github.com/Systems-Development-and-Frameworks/SCP-Foundation/pull/3

File: webapp/src/components/ListItem/ListItem.spec.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+                item: {id: '1', title: 'Foo', votes:0}
+            }
+        });
+    });
+    it('render item attributes', () => {
+        expect(wrapper.text()).toContain('Foo');
+        expect(wrapper.text()).toContain('0');
+    })
+    it('render upvote, downvote and delete buttons', () => {
+        expect(wrapper.text()).toContain('Upvote');
```
Comment (roschaefer) [inline webapp/src/components/ListItem/ListItem.spec.js:18]: You could make each assertion a separate test case.

File: webapp/src/components/BasicInput/BasicInput.stories.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import BasicInput from './BasicInput.vue';
+
+export default {
+    title: 'Example/BasicInput',
+    component: BasicInput,
```
Comment (roschaefer) [inline webapp/src/components/BasicInput/BasicInput.stories.js:5]: Adding this here:
```js
  argTypes: {
    'createItem': { action: 'create-item' }
  }
```

And then render your `BasicInput` like this:

```
const Template = (args, {argTypes}) => ({
    props: Object.keys(argTypes),
    components: {BasicInput},
    template:
        '<BasicInput @create-item="createItem"/>',
});
```

Will configure `Actions` tab:

Screenshot - 2020-11-11T211058 620

**Review by roschaefer — COMMENTED** [review]
fantastic

Congratulations! 10/10 :star: well done!

See my suggestions below. You don't have it included in this PR but your build server setup contains the following lines:
```
    strategy:
      matrix:
        os: [windows-latest, ubuntu-20.04]
        node-version: [10.x, 12.x, 14.x]
```

Since you need nodejs only to build VueJS, it's not necessary to test multiple versions of node. Same for the operating systems.

File: webapp/src/components/App/App.spec.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+                            votes: 1
+                        }
+                    ],
+                    descending: false,
+                }
+            }
+        });
+    });
+    it('does not render empty state', () => {
+        !expect(wrapper.text()).not.toContain('The list is empty :(');
```
[the remark this answers is not in the document]
Comment (MatzeK105) [reply webapp/src/components/App/App.spec.js:79]: My first thought was that it is a negotiation for the expect, but I tested it in our code and it was not. And that makes sense with your explanation. Thanks.

**Pull request merged**